test/e2e/plain_provisioner_test.go

```diff
@@ -882,9 +882,27 @@ var _ = Describe("plain provisioner bundleinstance", func() {
 
 			err = c.Create(ctx, biOriginal)
 			Expect(err).To(BeNil(), "failed to create original bundle instance")
-			Eventually(func() error { // Guarantee that the original bundle instance is create prior to the duplicate
-				return c.Get(ctx, client.ObjectKeyFromObject(biOriginal), &rukpakv1alpha1.BundleInstance{})
-			}).Should(Succeed(), "failed to validate original bundle instance was created in time")
+
+			// ensure the original BI owns the underlying bundle before creating the duplicate
+			By("projecting a successful installation status for the original BundleInstance")
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(biOriginal), biOriginal); err != nil {
+					return false
+				}
+				if biOriginal.Status.InstalledBundleName != bundle.GetName() {
+					return false
+				}
+				existing := meta.FindStatusCondition(biOriginal.Status.Conditions, rukpakv1alpha1.TypeInstalled)
+				if existing == nil {
+					return false
+				}
+				expected := metav1.Condition{
+					Type:   rukpakv1alpha1.TypeInstalled,
+					Status: metav1.ConditionStatus(corev1.ConditionTrue),
+					Reason: rukpakv1alpha1.ReasonInstallationSucceeded,
+				}
+				return conditionsSemanticallyEqual(*existing, expected)
+			}).Should(BeTrue())
```

Comment:
We've been building on top of this pattern that Joe had introduced recently: https://github.com/operator-framework/rukpak/blob/8f87c5cad26eb64d04516a5e65455c633f3e3aa7/test/e2e/plain_provisioner_test.go#L290-L297. I think my preference is using that pattern as it's slightly easier to ready, but this implementation is fine with me too.

Reply:
Yes, I think it's a nicer pattern -- but let's leave the original implementation in place for now and have a refactor as a potential followup.